io-engine/src/bdev/nexus/nexus_io.rs

```diff
@@ -489,34 +590,43 @@ impl<'n> NexusBio<'n> {
         // reset all I/O channels are de-initialized, so no I/O
         // submission is possible (spdk returns -6/ENXIO), so we have to
         // start device retire.
-        // TODO: ENOMEM and ENXIO should be handled differently and
+
+        // TODO:
+        // ENOMEM and ENXIO should be handled differently and
         // device should not be retired in case of ENOMEM.
         if result.is_err() {
             let device = failed_device.unwrap();
             // set the IO as failed in the submission stage.
-            self.ctx_mut().must_fail = true;
+            self.ctx_mut().failed += 1;
 
             self.channel_mut().disconnect_device(&device);
 
             self.fault_device(
                 &device,
                 IoCompletionStatus::IoSubmissionError(
-                    IoSubmissionFailure::Read,
+                    IoSubmissionFailure::Write,
                 ),
             );
         }
 
-        // partial submission
-        if inflight != 0 {
-            // An error was experienced during submission. Some IO however, has
-            // been submitted successfully prior to the error condition.
+        if inflight > 0 {
+            // TODO: fix comment:
+            // An error was experienced during submission.
+            // Some IO however, has been submitted successfully
+            // prior to the error condition.
             self.ctx_mut().in_flight = inflight;
             self.ctx_mut().status = IoStatus::Success;
-            self.ok_checked();
-            return result;
-        }
 
-        self.fail_checked();
+            // WAS: self.ok_checked();
```

Comment:
what does WAS mean?

Reply:
Past tense of "be" :-) It's a remark for myself to keep track of what I did, I forgot to remove it.